src/Common/tests/System/PlatformDetection.cs

```diff
@@ -206,5 +258,73 @@ private struct RTL_OSVERSIONINFOEX
             [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 128)]
             internal string szCSDVersion;
         }
+
+        private static int s_isWindowsElevated = -1;
+
+        public static bool IsWindowsAndElevated
+        {
+            get
+            {
+                if (s_isWindowsElevated != -1)
```

Comment:
nit: may be you can make s_isWindowsElevated as Nullable<bool> so you don't have to store -1 or other values on it.